### PR TITLE
Add consistency for the maximum_items_per_page option.

### PR DESCRIPTION
**Pull request by bpolaszek**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #3050 
| License       | MIT
| Doc PR        | api-platform/doc#874

This PR introduces a `pagination_maximum_items_per_page` as the new source of truth for hard-limiting the number of items in a resultset.

Former attribute `maximum_items_per_page` still works as a fallback with a deprecation notice when used.

### CODE REVIEW

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 1]
```diff
@@ -368,7 +368,26 @@ private function updateGetOperation(bool $v3, \ArrayObject $pathOperation, array
...
+                        ];
+
+                        $maxItemsPerPage = $resourceMetadata->getCollectionOperationAttribute($operationName, 'maximum_items_per_page', null, true);
+                        if (null !== $maxItemsPerPage) {
+                            @trigger_error('The "maximum_items_per_page" option has been deprecated since API Platform 2.5 in favor of "pagination_maximum_items_per_page" and will be removed in API Platform 3.', E_USER_DEPRECATED);
+                        }
+                        $maxItemsPerPage = $resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_maximum_items_per_page', $maxItemsPerPage, true);
+
+                        if (null !== $maxItemsPerPage) {
+                            $itemPerPageParameter['schema']['maximum'] = $maxItemsPerPage;
```
Comment (alanpoulain) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: It's a feature, right?

Comment (bpolaszek) [reply src/Swagger/Serializer/DocumentationNormalizer.php]: That's what I thought at first... but https://github.com/api-platform/core/issues/3050#issuecomment-530377865

**Comment by soyuka** [comment]
Should we merge this on 2.5? Some ci are failing?

**Comment by bpolaszek** [comment]
@soyuka Checked with @dunglas, I shouldn't have rebased on 2.4. 
Let this PR apart for the moment, I'll make another one based on 2.5 this morning.

**Comment by bpolaszek** [comment]
See #3088.

**Pull request closed without merging**